numpy/add_newdocs.py

```diff
@@ -1928,12 +1928,22 @@ def luf(lamdaexpr, *args, **kwargs):
     """
     dot(a, b, out=None)
 
-    Dot product of two arrays.
+    Dot product of two arrays. Specifically,
+
+    - If both `a` and `b` are 2-D arrays, it is matrix multiplication.
+
+    - If both `a` and `b` are 1-D arrays, it is inner product of vectors
+      (without complex conjugation).
+
+    - If either `a` or `b` is 0-D (scalar), it is equivalent to :func:`multiply`
+      (``numpy.multiply(a, b)``).
+
+    - If `a` is an N-D array and `b` is a 1-D array, it is a sum product over
+      the last axis of `a` and `b`.
 
-    For 2-D arrays it is equivalent to matrix multiplication, and for 1-D
-    arrays to inner product of vectors (without complex conjugation). For
-    N dimensions it is a sum product over the last axis of `a` and
-    the second-to-last of `b`::
+    - If `a` is an N-D array and `b` is an M-D array (where ``M>=2``), it is a
+      sum product over the last axis of `a` and the second-to-last axis of `b`,
+      e.g.,::
```

Comment:
Nit: `e.g.` seems slightly unnecessary in the presence of the `:` in the output.